src/Data/Array/Polarized.hs

```diff
@@ -129,7 +129,16 @@ import Prelude.Linear
 -- NOTE: this does NOT require allocation and can be in-lined.
 transfer :: Pull.Array a %1 -> Push.Array a
 transfer (Pull.Array f n) =
-  Push.Array (\k -> NonLinear.foldMap' (\x -> k (f x)) [0 .. (n - 1)])
+  -- 'transfer' was
+  -- > transfer (Pull.Array f n) =
+  -- >   Push.Array (\k -> NonLinear.foldMap' (\x -> k (f x)) [0 .. (n - 1)])
+  -- but 'Linear.Monoid' no longer implies 'NonLinear.Monoid'. So we can have
+  -- @mempty :: a@ and @(<>) :: a -> a -> a@ (by degrading 'Linear.<>'), but we
+  -- no longer have the 'NonLinear.Monoid' instance required to use
+  -- 'NonLinear.foldMap\''. As a result, we just expand 'foldMap\'' to its
+  -- definition in terms of 'foldl\'', which doesn't require 'NonLinear.Monoid':
+  -- > foldMap' f' = foldl' (\acc a -> acc <> f' a) mempty
```

Comment:
As a side note we still have that linear monoids have a corresponding classical monoid.

So we could have
```haskell
newtype Linear a = Linear a

instance Semigroup a => Linear.Semigroup (Linear a) where
  x <> y = x <>  y
instance Monoid a => Linear.Monoid (Linear a) where
  mempty = mempty
```

Then you could in principle do

```haskell
case NonLinear.Foldmap' (\x -> Linear (k (f x)) [0 .. (n - 1)] of
  Linear r -> r
```

It's not better than a simple `foldl'`, but it's worth being aware of the possibility (in an ideal world you would be able to use `coerce`, but `coerce` isn't linear).

---

There is also the option of using a linear `foldMap'`, but we don't have it available until we solve the Foldable issue (_e.g._ #190 ).